From 59baaeb6c223b96c432f256c044b04b2c704d63f Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Tue, 2 Feb 2016 10:22:39 -0500
Subject: [PATCH 12/23] Documentation: devicetree: bindings: arm: keystone:
 keystone-crypto.txt

Added device tree bindings for the side-band Security Accelerator
on keystone device.

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
---
 .../devicetree/bindings/soc/ti/keystone-crypto.txt | 68 ++++++++++++++++++++++
 1 file changed, 68 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/soc/ti/keystone-crypto.txt

diff --git a/Documentation/devicetree/bindings/soc/ti/keystone-crypto.txt b/Documentation/devicetree/bindings/soc/ti/keystone-crypto.txt
new file mode 100644
index 0000000..0194f0f
--- /dev/null
+++ b/Documentation/devicetree/bindings/soc/ti/keystone-crypto.txt
@@ -0,0 +1,68 @@
+Keystone Crypto Driver
+
+This document explains the device tree bindings for the side-band Security
+Accelerator on keystone devices.
+
+Required properties:
+ - compatible: should be "ti,netcp-sa-crypto".
+
+ - label: must be "crypto-gbe".
+
+ - clock: phandle to clksa clock.
+
+ - reg: Should contain register location and length of the SA hardware
+	module.
+
+ - ti,navigator-dmas and ti,navigator-dma-names: Security accelerator uses
+	2 dma channels. Rx and Tx. The first property is the pandle to the
+	knav_dmas subsystem and flow for RX channel and channel number for
+	TX channel. The second property is corresponding channel names.
+
+ - ti,rx-channel: name of RX channel.
+
+ - ti,rx-queue-depth and ti,rx-buffer-size: RX channel may use upto 4 free
+	descriptor queue. Those properties describes number of buffers in
+	each queue and the buffer's length.
+
+ - ti,rx-compl-queue: RX completion queue number.
+
+ - ti,rx-pool: number of buffer descriptos and region for rx-pool.
+
+ - ti,tx-channel: name of TX channel.
+
+ - ti,tx-queue-depth: number of TX descriptos.
+
+ - ti,tx-completion-queue: TX completion queue number.
+
+ - ti,tx-pool: number of buffer descriptos and region for tx-pool.
+
+ - ti,tx-submit-queue: submition queue number for TX channel.
+
+ - ti,sc-id: range of security contex IDs available for driver
+
+
+Example:
+
+crypto@20c0000 {
+	compatible = "ti,netcp-sa-crypto";
+	label = "crypto-gbe";
+	reg = <0x20c0000 0x40000>;
+	clocks = <&clksa>;
+
+	ti,navigator-dmas = <&dma_gbe 26>, <&dma_gbe 6>;
+	ti,navigator-dma-names = "crypto-rx", "crypto-tx";
+
+	ti,rx-channel = "crypto-rx";
+	ti,rx-queue-depth = <256 64 0 0>;
+	ti,rx-buffer-size = <1500 4096 0 0>;
+	ti,rx-compl-queue = <8712>;
+	ti,rx-pool	= <512 12>;
+
+	ti,tx-channel = "crypto-tx";
+	ti,tx-queue-depth = <256>;
+	ti,tx-completion-queue = <8713>;
+	ti,tx-pool = <512 12>;
+	ti,tx-submit-queue = <646>;
+
+	ti,sc-id = <0x7000 0x71ff>;
+};
-- 
1.9.1

